donut/modules/courses/templates/scheduler.html

```diff
@@ -461,7 +486,58 @@ <h1>Course Scheduler</h1>
     function displayUnits() {
       units.text('Units: ' + String(coursesUnits))
     }
-
+  $('#modal').on('show.bs.modal', function(e) {
+    var courseNum = $(e.relatedTarget).data("courseNum");
+    var courseId = $(e.relatedTarget).data("courseId");
+    displayNotes(courseNum, courseId); 
+  });
+  $('#editButton').click(function(e) {
+    $('#postNotes').show();
+    $('#viewNotes').hide();
+    $('#editButton').hide();
+  });
+  $('#postNotes').submit(function(e) {
+    e.preventDefault();
+    $.ajax({
+      url: '/1/scheduler/edit_notes',
+      type: 'POST',
+      data:$('#postNotes').serialize(),
+      success:function(){
+        displayNotes($('#courseNum').val(), $('#courseId').val());
+      },
+      error: function() {
+        displayWarning('Failed to save notes')
+      }
+    });
+  });
+  function displayNotes(courseNum, courseId) {
+    $('#courseId').val(courseId);
+    $('#courseNum').val(courseNum);
+    $('#modalTitle').text(courseNum);
+    $.ajax({
+      type: 'GET',
+      url: '/1/scheduler/notes/' + courseId,
+      success: function(notes) {
+        $('#viewNotes').html(URLify(notes));
+	$('#viewNotes').show();
+	$('#editButton').show();
+	$('#postNotes').hide();
+	$('#notes').text(notes);
+      },
+      error: function() {
+        displayWarning('Failed to load notes')
+      }
+    })
+  }
+  function URLify(string) {
+    const urls = string.match(/(((ftp|https?):\/\/)[\-\w@:%_\+.~#?,&\/\/=]+)/g);
+    if (urls) {
+      urls.forEach(function (url) {
+        string = string.replace(url, '<a target="_blank" href="' + url + '">' + url + "</a>");
+      });
+    }
+    return string.replace("(", "<br/>(");
+  }
```

Comment:
Use `format-text.js` instead? Injecting unsanitized strings as HTML is not a great practice, even if users can only do it to themselves